biz.aQute.bndlib/src/aQute/bnd/exporter/executable/ExecutableJarExporter.java

```diff
@@ -28,26 +41,48 @@ public ExecutableJarExporter() {}
 	@Override
 	public String[] getTypes() {
 		return new String[] {
-			EXECUTABLE_JAR
+			EXECUTABLE_JAR, EXECUTABLE_PACK
 		};
 	}
 
 	@Override
 	public Entry<String, Resource> export(String type, Project project, Map<String, String> options) throws Exception {
-		Configuration configuration = Converter.cnv(Configuration.class, options);
 		project.prepare();
-		try (ProjectLauncher launcher = project.getProjectLauncher()) {
-			launcher.setKeep(configuration.keep());
-			Jar jar = launcher.executable();
-			project.getInfo(launcher);
-
-			String name = jar.getName();
-			String[] baseext = Strings.extension(name);
-			if (baseext != null && ("bnd".equals(baseext[1]) || "bndrun".equals(baseext[1]))) {
-				name = baseext[0];
+		Jar jar;
+		if (EXECUTABLE_JAR.equals(type)) {
+			try (ProjectLauncher launcher = project.getProjectLauncher()) {
+				//
+				// Ensure that we do not override the -runkeep property
+				// if keep is not explicitly set. We therefore take the `or`
+				// of the keep option and the -runkeep
+				//
+
+				Configuration configuration = Converter.cnv(Configuration.class, options);
+				if (configuration.keep() || Processor.isTrue(project.getProperty(Constants.RUNKEEP)))
+					launcher.setKeep(true);
+
+				jar = launcher.executable();
+				launcher.removeClose(jar);
+				project.getInfo(launcher);
 			}
-			name = name + Constants.DEFAULT_JAR_EXTENSION;
-			return new SimpleEntry<>(name, new JarResource(jar, true));
+		} else if (EXECUTABLE_PACK.equals(type)) {
+
+			String profile = options.getOrDefault(Constants.PROFILE, project.getProperty(Constants.PROFILE));
```

Comment:
There is a `Processor.getProfile()` method which may make sense for the default value.

Although this method has an odd implementation and no one seems to use it.

Reply:
k